core/scripts/cre/environment/environment/environment.go

```diff
@@ -292,7 +296,7 @@ func startCmd() *cobra.Command {
 				return errors.Wrap(err, "either cron binary path must be set in TOML config (%s) or you must use Docker image with all capabilities included and passed via withPluginsDockerImageFlag")
 			}
 
-			output, startErr := StartCLIEnvironment(cmdContext, in, topology, withPluginsDockerImage, defaultCapabilities, capabilityFlagsProvider)
+			output, startErr := StartCLIEnvironment(cmdContext, in, topology, withPluginsDockerImage, withContractsVersion, defaultCapabilities, capabilityFlagsProvider)
```

Comment:
would it be a blasphemy to follow the same pattern we do with capability flags provider? lift out contract set creation based on the flag and pass contract set to `StartCLIEnvironment` function